zerver/lib/actions.py

```diff
@@ -7172,11 +7172,15 @@ def user_info(um: UserMessage) -> Dict[str, Any]:
                 # immediate succession; this is correct only because
                 # muted_topics events always send the full set of topics.
                 remove_topic_mute(muting_user, stream_being_edited.id, orig_topic_name)
-                do_mute_topic(
-                    muting_user,
-                    new_stream if new_stream is not None else stream_being_edited,
-                    topic_name if topic_name is not None else orig_topic_name,
-                )
+                try:
```

Comment:
We could use `topic_is_muted` here too, but since we were using `try catch` elsewhere for this, I used it. The no of database queries is the same for both.

Reply:
I don't think we can add a transaction here safely; `do_update_message` may already be in a transaction.

Reply:
Tests are failing without it otherwise it works fine with or without `transation.atomic`. What do you suggest we do?

Reply:
Ok, I think it is best to just do `topic_is_muted` check here, right?